### PR TITLE
WOS-575 Quick start for developers revision + add docs about end-to-end testing

### PR DESCRIPTION
**Pull request by Mikilll94**
- Add docs about end-to-end functional testing with Cypress
- Add docs about end-to-end visual regression testing with BackstopJS
- Remove incorrect and not realistic in normal dev workflow use of Cypress in "Quick start for developers"

### CODE REVIEW

File: docs/cms/developers/e2e-testing/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# End-to-end testing
+
+## Introduction
+
+**End-to-end testing (E2E testing)** checks if the software is working correctly from the end user point of view. It verifies all the flows and actions which the user may perform when using the application. E2E tests should be executed in a real-world environment like a web browser to ensure that the testing closely resembles the actual user experience.
+
+## Test content
+
+To run tests in the context of CMS we need to decide what `content` created by authors we will use. In Websight CMS for testing purposes we can use the content from the following folders:
```
Comment (paweljozwicki) [inline docs/cms/developers/e2e-testing/index.md:9]: I'm not sure if we should write that: "In Websight CMS for testing purposes we can use the content from the following folders" - content placed in such modules is related only to the project structure that we choose. It's not enforced by WebSight itself. 
Maybe we should put some link to the starter project and we could write all this page in the starter context. 
I mean, we could write that we can write e2e test for our project and in the sample Starter project we have functional and visual tests. For functional test purposes, we created content in the /tests/content module.... And that we prepared visual tests for the /content to ensure that our initial content won't break....

File: docs/cms/developers/e2e-testing/visual-regression-testing/index.md
Comment (paweljozwicki) [inline docs/cms/developers/e2e-testing/visual-regression-testing/index.md:1]: Let's add information that in the Starter project we used BackstopJS

File: docs/cms/developers/e2e-testing/functional-testing/index.md
Comment (paweljozwicki) [inline docs/cms/developers/e2e-testing/functional-testing/index.md:1]: Let's add information that in the Starter project we used Cypress

File: docs/cms/developers/e2e-testing/index.md (hunk 1 above)
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/index.md]: done

File: docs/cms/developers/e2e-testing/visual-regression-testing/index.md
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/visual-regression-testing/index.md:1]: done

File: docs/cms/developers/e2e-testing/functional-testing/index.md
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/functional-testing/index.md:1]: done

**Review by paweljozwicki — APPROVED** [review]

File: docs/cms/developers/e2e-testing/index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# End-to-end testing
+
+## Introduction
+
+**End-to-end testing (E2E testing)** checks if the software is working correctly from the end user point of view. It verifies all the flows and actions which the user may perform when using the application. E2E tests should be executed in a real-world environment like a web browser to ensure that the testing closely resembles the actual user experience.
+
+While you can write your own E2E tests using your favorite tools you can also leverage our solution from [starter](https://github.com/websight-io/starter). In this project we implemented the following types of E2E tests:
```
[the remark this answers is not in the document]
Comment (kamil-kwarciak) [reply docs/cms/developers/e2e-testing/index.md:7]: not sure if we used the term started in the docs before. we should clarify what it is

**Review by kamil-kwarciak — CHANGES_REQUESTED** [review]
see my comments and adjust the docs

File: docs/cms/developers/e2e-testing/index.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- [Visual regression tests with BackstopJS](/cms/developers/e2e-testing/visual-regression-testing)
+
+## Test content
+
+To run tests in the context of CMS we need to decide what content we will use. In [starter](https://github.com/websight-io/starter) during testing we have configured to use content from the following folders:
+
+- `/content` - This is the initial content for CMS. It is a good place to keep the actual content used by the end user.
+- `/tests/content` - This is the content which is only available during E2E testing. Here we can store the content which was crafted specifically for the purpose of testing, i.e. we can use this folder to test CMS components in isolation.
+
+In your tests you can use `/content` and `/tests/content` folders simultanously.
```
[the remark this answers is not in the document]
Comment (kamil-kwarciak) [reply docs/cms/developers/e2e-testing/index.md:19]: Discussed. no action needed.

File: docs/cms/developers/e2e-testing/index.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+While you can write your own E2E tests using your favorite tools you can also leverage our solution from [starter](https://github.com/websight-io/starter). In this project we implemented the following types of E2E tests:
+
+- [Functional tests with Cypress](/cms/developers/e2e-testing/functional-testing)
+- [Visual regression tests with BackstopJS](/cms/developers/e2e-testing/visual-regression-testing)
+
+## Test content
+
+To run tests in the context of CMS we need to decide what content we will use. In [starter](https://github.com/websight-io/starter) during testing we have configured to use content from the following folders:
+
+- `/content` - This is the initial content for CMS. It is a good place to keep the actual content used by the end user.
```
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/index.md:16]: nope, this is not the place for user content. This is just the initial content of CMS. You can place any content there. So phrase "good place" is not bad.

File: docs/cms/developers/e2e-testing/visual-regression-testing/index.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Visual regression testing with BackstopJS
+
+![BackstopJS logo](backstopjs-logo.png)
+
+## Introduction
+
+**Visual regression testing** is used to detect unintended visual changes in web application. It involves capturing screenshots of web pages or components at different points in time and comparing them to detect any differences. These differences needs to be reviewed if they are expected or not.
```
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/visual-regression-testing/index.md]: done

File: docs/cms/developers/e2e-testing/functional-testing/index.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+# Functional testing with Cypress
+
+![Cypress logo](cypress-logo.png)
+
+## Introduction
+
+**Cypress** is a modern open-source end-to-end testing framework which provides good developer experience. You can check how we leverage Cypress in [starter](https://github.com/websight-io/starter) to perform **functional testing**. Its goal is to check if the end user can complete all the flows and actions available on the website.
```
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/functional-testing/index.md]: done

File: docs/cms/developers/e2e-testing/index.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# End-to-end testing
+
+## Introduction
+
+**End-to-end testing (E2E testing)** checks if the software is working correctly from the end user point of view. It verifies all the flows and actions which the user may perform when using the application. E2E tests should be executed in a real-world environment like a web browser to ensure that the testing closely resembles the actual user experience.
```
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/index.md]: done

File: docs/cms/developers/e2e-testing/index.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+While you can write your own E2E tests using your favorite tools you can also leverage our solution from [starter](https://github.com/websight-io/starter). In this project we implemented the following types of E2E tests:
+
+- [Functional tests with Cypress](/cms/developers/e2e-testing/functional-testing)
+- [Visual regression tests with BackstopJS](/cms/developers/e2e-testing/visual-regression-testing)
+
+## Test content
+
+To run tests in the context of CMS we need to decide what content we will use. In [starter](https://github.com/websight-io/starter) during testing we have configured to use content from the following folders:
```
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/index.md:18]: This requires a separate article about Maven configuration.

File: docs/cms/developers/e2e-testing/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mikilll94) [reply docs/cms/developers/e2e-testing/index.md:7]: Added more info about `starter` project in "Quick start for developers". Added references to `starter` in end-to-end docs.

File: docs/cms/developers/quick-start/index.md [hunk 9]
```diff
@@ -2,9 +2,19 @@
...
 
-This guide explains how to update a simple Howlite component. Before you complete this guide, we recommend completing the basic [quick start guide](/cms/quick-start/) first so that you understand basic WebSight CMS concepts.
+!!! info "Important notice"
+
+    Before you complete this guide, we recommend completing the basic [quick start guide](/cms/quick-start/) first so that you understand basic WebSight CMS concepts.
+
+This guide uses [starter](https://github.com/websight-io/starter) - our official project which demonstrates variuous functionality of Websight CMS and pre-configures for you the following features:
+
+- Demo site project _Luna_
+- Integration with [Howlite](https://github.com/websight-io/howlite) library
```
Comment (kamil-kwarciak) [inline docs/cms/developers/quick-start/index.md:12]: Integration is not the right wording here

File: docs/cms/developers/quick-start/index.md [hunk 10]
```diff
@@ -2,9 +2,19 @@
 
 ## Introduction
 
-This guide explains how to update a simple Howlite component. Before you complete this guide, we recommend completing the basic [quick start guide](/cms/quick-start/) first so that you understand basic WebSight CMS concepts.
+!!! info "Important notice"
+
+    Before you complete this guide, we recommend completing the basic [quick start guide](/cms/quick-start/) first so that you understand basic WebSight CMS concepts.
+
+This guide uses [starter](https://github.com/websight-io/starter) - our official project which demonstrates variuous functionality of Websight CMS and pre-configures for you the following features:
```
Comment (kamil-kwarciak) [inline docs/cms/developers/quick-start/index.md:9]: our official -> a sample ?

File: docs/cms/developers/quick-start/index.md (hunk 9 above)
Comment (Mikilll94) [reply docs/cms/developers/quick-start/index.md]: changed this to "Howlite components library"

File: docs/cms/developers/quick-start/index.md (hunk 10 above)
Comment (Mikilll94) [reply docs/cms/developers/quick-start/index.md]: done

**Review by kamil-kwarciak — APPROVED** [review]

**Pull request merged**